packages/components/select-menu/src/select-menu.js

```diff
@@ -149,8 +171,8 @@ export class TSSelectMenu extends TSElement {
 				: html`
 				<ul>
 					${
-						filteredItems.length > 0
-							? filteredItems.map(
+						this.displayedItems.length > 0
+							? this.displayedItems.map(
```

Comment:
You call getter of `displayedItems` twice, but it has a complex logic inside. Consider to call it once and store the value in a local variable.